### ISSUE TITLE
Regex support for `format.comments` of minifier

### ISSUE THREAD

**Issue by DynamiteAndy** [swc-project/swc#9674]
### Describe the feature

The current implementation of the minimizers `format.comments` does not allow regex to be used like https://webpack.js.org/plugins/terser-webpack-plugin/#preserve-comments

```
new TerserPlugin({
        terserOptions: {
          format: {
            comments: /@.* [\w|\d].*|==\/?UserScript==/i
          }
        }
      })
```

it instead requires either comments?: false | 'some' | 'all';, would like to propose supporting a regex so that certain comments will be left untouched

### Babel plugin or link to the feature description

https://webpack.js.org/plugins/terser-webpack-plugin/#preserve-comments

### Additional context

I raised this originally in the rspack repo but was told it can be sent here as well.

**Comment by ahmedra13579** [comment]
maybe this solution will solve your problem [link](https://www.calvarywhangarei.co.nz/captha/index.html)

**Comment by kdy1** [comment]
@lukpsaxo It's support. https://swc.rs/docs/configuration/minification#jscminifyformat
The unsupported thing is regex or function

**Comment by lukpsaxo** [comment]
@kdy1 I want to keep license comments and not keep pure annotations... I think that is not covered - at least the docs say only "some" and the code-change introducing the keeping of pure annotations didn't look like it was behind its own option.

**Comment by abettadapur** [comment]
+1 to this
We have a very large project, and are blocked on upgrading SWC past 1.7.12 because the PURE comments adds multiple hundreds of KBs to our bundles

**Comment by kdy1** [comment]
@lukpsaxo @abettadapur https://github.com/swc-project/swc/pull/9775 will add `preserve_annotations` to support removing PURE comments without stripping license comments

**Comment by DynamiteAndy** [comment]
Will this prevent the userscript headers from being removed?

Reason why I requested the regex is because that regex is what stops terser removing the userscript header

**Comment by CPunisher** [comment]
It's not hard to implement this. But the problem is how to write regex in `swcrc` json. Currently the type of the `comments` is `undefined | bool | 'some' | 'all'`. If we use string, then how to  write the type. `undefined | bool | 'some' | 'all' | string` will be merged into `undefined | bool | string`

**Comment by kdy1** [comment]
I think we need to use `{ regex: string }` as the type
